docs/release_process.md

```diff
@@ -0,0 +1,25 @@
+# Release process
+
+## Frequency
+
+Open Service Mesh will target a release every quarter. These releases will be targeted in the first half of February, May, August, and November. These target release months are subject to change if a release is pushed. In other words, if a release misses February and is pushed out in March then the next release will target June instead of May.
```

Comment:
The release guide has a section regarding release candidates that I think would fit better in this doc: https://github.com/openservicemesh/osm/blob/edd9ea76246f8f9f673de2db30be1eadef232a18/docs/release_guide.md?plain=1#L13-L19

Then we'd have a clear line between **when** to cut a release (this doc) vs. **how** (the release guide).

Reply:
Just be clear, @nojnhuh, you're recommending that we move that `Release Candidates` section from `release_guide.md` to `release_process.md` (the new file in this PR)?

Reply:
Yeah. We could of course change the language to make it more consistent with the rest of this doc, but I think that information would fit better here than in the release guide.

Reply:
Agreed. Done!